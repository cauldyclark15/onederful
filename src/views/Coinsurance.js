import React from 'react';
import PropTypes from 'prop-types';
import { Card, Table } from 'antd';

const propTypes = {
  coinsurance: PropTypes.array,
};

const defaultProps = {
  coinsurance: [],
};

function Coinsurance({ coinsurance }) {
  const dataSource = coinsurance.map((item, index) => ({
    ...item,
    key: index,
  }));
  const columns = [
    {
      title: 'Insurance Type',
      dataIndex: 'insurance_type',
    },
    {
      title: 'Service Type',
      dataIndex: 'service_type',
    },
    {
      title: 'Network Status',
      dataIndex: 'in_network_status',
    },
    {
      title: '% String',
      dataIndex: 'percentage_string',
    },
    {
      title: '% for Provider',
      dataIndex: 'percentage_for_provider',
    },
    {
      title: 'Percentage',
      dataIndex: 'percentage',
    },
  ];
  return (
    <Card title="Coinsurance" style={{ width: '100%', height: '100%' }}>
      <Table columns={columns} dataSource={dataSource} />
    </Card>
  );
}

Coinsurance.propTypes = propTypes;
Coinsurance.defaultProps = defaultProps;

export default Coinsurance;
