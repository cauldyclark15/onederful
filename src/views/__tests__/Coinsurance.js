import React from 'react';
import renderer from 'react-test-renderer';

import Coinsurance from '../Coinsurance';

it('renders correctly and unchange so far', () => {
  const tree = renderer.create(<Coinsurance />).toJSON();

  expect(tree).toMatchSnapshot();
});
