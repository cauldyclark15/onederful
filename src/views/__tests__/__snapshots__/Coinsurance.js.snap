// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and unchange so far 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      <div
        className="ant-card-head-title"
      >
        Coinsurance
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-table-wrapper"
    >
      <div
        className="ant-spin-nested-loading"
        style={null}
      >
        <div
          className="ant-spin-container"
        >
          <div
            className="ant-table ant-table-default ant-table-empty ant-table-scroll-position-left"
            style={Object {}}
          >
            <div
              className="ant-table-content"
            >
              <div
                className="ant-table-body"
                onScroll={[Function]}
                onWheel={[Function]}
                style={Object {}}
              >
                <table
                  className=""
                  style={Object {}}
                >
                  <colgroup>
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <col
                      style={
                        Object {
                          "minWidth": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </colgroup>
                  <thead
                    className="ant-table-thead"
                  >
                    <tr
                      style={
                        Object {
                          "height": null,
                        }
                      }
                    >
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Insurance Type
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Service Type
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Network Status
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          % String
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          % for Provider
                        </div>
                      </th>
                      <th
                        className=""
                      >
                        <div
                          onClick={[Function]}
                        >
                          Percentage
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    className="ant-table-tbody"
                  />
                </table>
              </div>
              <div
                className="ant-table-placeholder"
              >
                No data
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
